kernel/src/log_segment.rs

```diff
@@ -1,22 +1,31 @@
 //! Represents a segment of a delta log. [`LogSegment`] wraps a set of  checkpoint and commit
 //! files.
 
-use crate::actions::{get_log_schema, Metadata, Protocol, METADATA_NAME, PROTOCOL_NAME};
-use crate::path::ParsedLogPath;
-use crate::schema::SchemaRef;
-use crate::{DeltaResult, Engine, EngineData, Error, Expression, ExpressionRef, FileMeta};
-
+use crate::{
+    actions::{get_log_schema, Metadata, Protocol, METADATA_NAME, PROTOCOL_NAME},
+    schema::SchemaRef,
+    DeltaResult, Engine, EngineData, Error, ExpressionRef, FileMeta,
+};
+use crate::{
+    path::ParsedLogPath, snapshot::CheckpointMetadata, utils::require, Expression,
+    FileSystemClient, Version,
+};
```

Comment:
nit: i think the large braces in imports actually worsen readability - can we revert?